src/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -0,0 +1,90 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Loader;
+using Internal.IO;
+
+namespace System
+{
+    internal class StartupHookProvider
+    {
+        private static string initializeMethodName = "Initialize";
+
+        // Parse a string specifying a list of assemblies and types
+        // containing a startup hook, and call each hook in turn.
+        private static void ProcessStartupHooks(IntPtr startupHooksVariable)
+        {
+            Debug.Assert(startupHooksVariable != IntPtr.Zero);
+            string[] startupHooks = Marshal.PtrToStringUTF8(startupHooksVariable).Split(new char[] { Path.PathSeparator });
+
+            // Parse startup hook variable
+            List<Tuple<string, string>> startupHooksParsed = new List<Tuple<string, string>>();
```

Comment:
Consider the C# 7 tuple syntax: `var startupHooksParsed = new List<(string Path, string Type)>();`

Reply:
Once you do that, you can use actual names instead of the opaque Item1 and Item2:
```csharp
foreach (var startupHook in startupHooksParsed)
{
    .... startupHook.Path ...
}
```

Reply:
Will do, thanks!